testplan/common/remote/ssh_client.py

```diff
@@ -0,0 +1,230 @@
+        :type key_path: ``str`` or ``NoneType``
+        """
+        self.host = host
+        self.port = port
+        self.logger = logger or TESTPLAN_LOGGER
+
+        self.paramiko_args = DEFAULT_PARAMIKO_CONFIG
+        self.paramiko_args.update(
+            {
+                "hostname": self.host,
+                "port": self.port,
+            }
+        )
+        self.paramiko_args.update(args)
+
+        self._ssh_client = None
+        self._sftp_client = None
+
+    def connect(self):
+        """
+        Establish an SSH connection.
+
+        :return: Self for method chaining
+        :rtype: ``SSHClient``
+        """
+        if self._ssh_client is not None:
+            self.logger.warning("SSH connection already established")
+            return self._ssh_client
+
+        ssh_client = paramiko.SSHClient()
+        ssh_client.set_missing_host_key_policy(paramiko.MissingHostKeyPolicy())
+
+        ssh_client.connect(**self.paramiko_args)
+        self.logger.debug(
+            "Connected to %s@%s:%s",
+            self.paramiko_args["username"],
+            self.paramiko_args["hostname"],
+            self.paramiko_args["port"],
+        )
+
+        self._ssh_client = ssh_client
+        return self._ssh_client
+
+    def exec_command(
+        self,
+        cmd: str | List[str],
+        label: str = None,
+        check: bool = True,
+        env: Dict = None,
+        timeout: int = 30,
+    ):
+        """
+        Run a command on the remote host.
+
+        :param cmd: Command to execute (either a string or list of arguments)
+        :type cmd: ``str`` or ``List[str]``
+        :param label: Label for identifying the command in logs (defaults to hash of command)
+        :type label: ``str`` or ``NoneType``
+        :param check: If True, raises exception when command fails
+        :type check: ``bool``
+        :param env: Environment variables to set for the command
+        :type env: ``Dict`` or ``NoneType``
+        :param timeout: Timeout for command execution in seconds
+        :type timeout: ``int``
+        :return: Tuple of (exit_code, stdout_str, stderr_str)
+        :rtype: ``tuple`` of (``int``, ``str``, ``str``)
+        :raises: ``RuntimeError`` if command fails and check is True
+        """
+        if not self._ssh_client:
+            self.connect()
+
+        if isinstance(cmd, list):
+            cmd = [str(a) for a in cmd]
+            # for logging, easy to copy and execute
+            cmd_string = " ".join(map(shlex.quote, cmd))
+        else:
+            cmd_string = cmd
+
+        if not label:
+            label = hash(cmd_string) % 1000
+
+        self.logger.debug(
+            "ssh_client executing command [%s]: '%s'",
+            label,
+            cmd_string,
+        )
+        start_time = time.time()
+
+        _, stdout, stderr = self._ssh_client.exec_command(
+            command=cmd_string,
+            timeout=timeout,
+            environment=env,
```

Comment:
`env` usually won't work if passed in this way, need to prepend them to command line instead